src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -954,7 +944,7 @@ public static unsafe int IndexOfAny(ref char searchStart, char value0, char valu
 
         VectorCompare:
             // We include the Supported check again here even though path will not be taken, so the asm isn't generated if not supported.
-            if (!Sse2.IsSupported && Vector.IsHardwareAccelerated)
+            if (!Vector128.IsHardwareAccelerated && Vector.IsHardwareAccelerated)
```

Comment:
Do we have any platforms which accelerate `Vector<T>` but not `Vector128<T>`?

This certainly doesn't exist for RyuJIT and I don't think that's true for Mono anymore either (CC. @fanyang-mono?)

If that's the case, we can probably just remove this path entirely.

Reply:
> If that's the case, we can probably just remove this path entirely.

Awesome, then once all my SpanHelpers PRs get merged I am going to send a new one that removes all `Vector<T>` code paths for the methods that are supporting `Vector128` now.

Reply:
We do accelerate both `Vector<T>` and `Vector128<T>` on both Arm64 and Amd64. On Arm64, the intrinsics support for their methods is complete now. However, it is not on Amd64. This issue tracks the progress (#66392).